### ISSUE TITLE
MainActivity restarts when clicking on Push notification that contains a uri

### ISSUE THREAD

**Issue by vgujare-rbi** [adobe/aepsdk-messaging-android#219]
### Expected Behaviour

Clicking on a push notification with a `uri` opens the existing `MainActivity` and passes the `uri` to it.

### Actual Behaviour

Clicking on a push notification with a `uri` restarts the `MainActivity` on Android leading to unexpected behaviour in the app.

#### Steps to Reproduce

- Incorporate the react native messaging sdk in a react-native app with the package versions mentioned below.
- Handle push notification display and tracking using `handleRemoteMessage` method from `MessagingService`.
- Using Adobe Journey Optimizer, send a push notification with a deeplink to the app.
- Keep the app open and click on the displayed notification. 
- Notice the `MainActivity` restarts which leads to the application not being ready to handle the `uri` passed to it by the intent as the app is still booting up.
- This issue is not observed if the Push notification click behaviour is set to simply open the app instead.

#### Platform and Version

- Platform: React-native (Expo Managed) - Android
- React native SDK: @adobe/react-native-aepmessaging: 5.0.0
- Native SDK: com.adobe.marketing.mobile:messaging: 2.2.0

#### Sample Code that illustrates the problem
I have created a `NotificationService` that gets the notification from firebase and passes it to the `handleRemoteMessage` method for display:

**`NotificationService.java`**
```java
public class NotificationService extends FirebaseMessagingService {
  ...
  @Override
  public void onMessageReceived(RemoteMessage message) {
    MessagingService.handleRemoteMessage(this, message);
  }
}
``` 
--- 

In my Manifest, I have: 

**`AndroidManifest.xml`**
```xml
<uses-permission android:name="android.permission.POST_NOTIFICATIONS"/>
...
<service android:name=".NotificationService" android:enabled="true" android:exported="false">
   <intent-filter>
      <action android:name="com.google.firebase.MESSAGING_EVENT"/>
   </intent-filter>
</service>
<activity android:name=".MainActivity" 
   android:label="@string/app_name" 
   android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode" 
   android:launchMode="singleTask"
   android:exported="true"
   android:screenOrientation="portrait">
   <intent-filter>
      <action android:name="android.intent.action.MAIN"/>
      <category android:name="android.intent.category.LAUNCHER"/>
   </intent-filter>
   <intent-filter>
      <action android:name="android.intent.action.VIEW"/>
      <category android:name="android.intent.category.DEFAULT"/>
      <category android:name="android.intent.category.BROWSABLE"/>
      <data android:scheme="testApp"/>
      <data android:scheme="com.testapp.testPackageName"/>
   </intent-filter>
....
</activity>
``` 
--- 

Upon digging through the Android SDK, I found that `MessagingPushTrackerActivity` has an [`openUri`](https://github.com/adobe/aepsdk-messaging-android/blob/main/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingPushTrackerActivity.java#L138) method that creates a `TaskStackBuilder`:

**`MessagingPushTrackerActivity.java`**
```java
private void openUri(final String uri) {
   final Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(uri));
   TaskStackBuilder stackBuilder = TaskStackBuilder.create(this);
   stackBuilder.addNextIntentWithParentStack(intent);
   stackBuilder.startActivities();
}
``` 

I believe the `TaskStackBuilder` recreates the task stack thereby restarting the existing `MainActivity`.

**Comment by rymorale** [comment]
Hi @PravinPK, would you be able to take a look at this? Also adding @dsoffiantini in case this issue is specific to react.

**Comment by vgujare-rbi** [comment]
> @vgujare-rbi Thank you for contacting us. The bug has been fix in Messaging 3.0.0, scheduled for release in the upcoming weeks. Will keep you posted.
> 
> Thanks, Pravin

That's wonderful news! Thank you so much for looking into this!

**Comment by spoorthipujariadobe** [comment]
@vgujare-rbi Messaging 3.0.0 has been released. Please close try out the latest version of the SDK and close this ticket if its fixed.
